### PR TITLE
fix regression: importing papagayo file didn't show the expected layers

### LINKED ISSUE #1804: Papagayo integration is buggy with synfig v1.3.16
Synfig v1.3.16 on windows 10 20H2

.pgo layer should show all the phonemes layers as its child layer after importing it, but it's not showing.
Screenshot 2020-10-25 203116

### PR DESCRIPTION
**Pull request by rodolforg**
Reported-by: BigBang
https://forums.synfig.org/t/no-group-for-the-tones-of-papagayao/12849

fix #1804

guilty commit: f63ade3652fa6d9d41315c227f71cf2ec764e9ec

Test file: [test_pgo.zip](https://github.com/synfig/synfig/files/7814071/test_pgo.zip)

### CODE REVIEW

**Comment by rodolforg** [comment]
The problem exists since 1.3.12 :O

**Comment by ice0** [comment]
@rodolforg 
Can you describe why it didn't work?

**Comment by rodolforg** [comment]
> Can you describe why it didn't work?

The layer parameter value stored and returned by `layer->get_params()` depends on current layer time set.
Sometimes it is not set yet, and I didn't want to force it.

Looking it again, in the case I added, we don't need to use `layer_switch->get_possible_new_layers(possible_new_layers);` as we already got them :)

**Comment by rodolforg** [comment]
Example file (copied from the forums):
```
lipsync version 1
test-sound.wav
60
243
1
	Voice 1
	that's a test
	1
		that's a test
		34
		110
		3
			that's 34 68 4
				34 etc
				43 AI
				52 etc
				60 etc
			a 69 76 1
				69 AI
			test 77 110 4
				77 etc
				86 E
				94 etc
				103 etc
```

File: synfig-studio/src/gui/trees/layertreestore.cpp [hunk 1]
```diff
@@ -894,7 +894,15 @@ LayerTreeStore::set_row_layer(Gtk::TreeRow &row, const synfig::Layer::Handle &ha
...
 			if (etl::handle<Layer_Switch> layer_switch = etl::handle<Layer_Switch>::cast_dynamic(handle))
 			{
-				if (!layer_switch->get_param("layer_name").get(String()).empty())
+				// is this Switch operated by layer-name or by depth-index?
+				bool does_param_layer_name_change = false;
+				auto iter = layer_switch->dynamic_param_list().find("layer_name");
+				if (iter != layer_switch->dynamic_param_list().end()) {
+					std::set<ValueBase> values;
+					iter->second->get_values(values);
+					does_param_layer_name_change = true;
```
Comment (ice0) [inline synfig-studio/src/gui/trees/layertreestore.cpp:903]: This part seems incomplete to me. 
Maybe this is debugging code? If yes, just comment it for future references:
```
// debug values
// std::set<ValueBase> values;
// iter->second->get_values(values);
```

or maybe it should be something like this?
```
std::set<ValueBase> values;
iter->second->get_values(values);
does_param_layer_name_change = values.size() > 1;
```

Comment (rodolforg) [reply synfig-studio/src/gui/trees/layertreestore.cpp:903]: Indeed.
I'll fix in other way, however.
I notice there are useless calls in my code  :) Thank you

Comment (rodolforg) [reply synfig-studio/src/gui/trees/layertreestore.cpp]: Done.

**Comment by rodolforg** [comment]
Now I avoid the repetition of `dynamic_cast()` on each (selected) layer parameter.

And there is no need to check the `layer_name` parameter value(s):
1. `layer_switch->get_possible_new_layers(possible_new_layers)` already check its values
2. this code just show the possible (and existent) layer children: highlighting what layer is currently selected is done elsewhere. (`LayerTreeStore::get_value_vfunc()`)

**Review by ice0 — APPROVED** [review]

**Comment by morevnaproject** [comment]
Hi! I have finally tested the PR and it works perfectly! Thank you so much!

**Review by morevnaproject — APPROVED** [review]

**Comment by morevnaproject** [comment]
Backported to 1.4.3.

**Pull request merged**